pkg/sql/create_view.go

```diff
@@ -92,25 +91,29 @@ func (n *createViewNode) startExec(params runParams) error {
 	// First check the backrefs and see if any of them are temporary.
 	// If so, promote this view to temporary.
 	backRefMutables := make(map[descpb.ID]*tabledesc.Mutable, len(n.planDeps))
+	hasTempBackref := false
 	for id, updated := range n.planDeps {
 		backRefMutable := params.p.Descriptors().GetUncommittedTableByID(id)
 		if backRefMutable == nil {
 			backRefMutable = tabledesc.NewExistingMutable(*updated.desc.TableDesc())
 		}
-		if !persistence.IsTemporary() && backRefMutable.Temporary {
-			// This notice is sent from pg, let's imitate.
-			params.p.BufferClientNotice(
-				params.ctx,
-				pgnotice.Newf(`view "%s" will be a temporary view`, viewName),
-			)
-			persistence = tree.PersistenceTemporary
+		if !n.persistence.IsTemporary() && backRefMutable.Temporary {
+			hasTempBackref = true
```

Comment:
Note: it seemed to me that the notice could end up being printed multiple times so I rewrote this.